services/node_create.proto

```diff
@@ -27,62 +25,111 @@ option java_multiple_files = true;
 
     /**
-     * Ip address and port, mandatory field. Fully qualified domain name is
-     * not allowed here. Maximum number of these endpoints is 10.
-     * The first in the list is used as the Internal IP address in config.txt,
-     * the second in the list is used as the External IP address in config.txt,
-     * the rest of IP addresses are ignored for DAB phase 2.
+     * A list of service endpoints for gossip.
+     * <p>
+     * These endpoints SHALL represent the published endpoints to which other
+     * consensus nodes may _gossip_ transactions.<br/>
+     * This list MUST NOT be empty.<br/>
+     * This list MUST NOT contain more than `10` entries.<br/>
+     * The first two entries in this list SHALL be the endpoints published to
+     * all consensus nodes.<br/>
+     * All other entries SHALL be reserved for future use.
+     * <p>
+     * Each network may have additional requirements for these endpoints.
+     * A client MUST check network-specific documentation for those
+     * details.<br/>
+     * <blockquote>Example<blockquote>
+     * Hedera Mainnet _requires_ that address be specified, and does not
+     * permit DNS name (FQDN) to be specified.<br/>
+     * Mainnet also requires that the first entry be an "internal" IP
+     * address and the second entry be an "external" IP address.
+     * </blockquote>
+     * <blockquote>
+     * Solo, however, _requires_ DNS name (FQDN) but also permits
+     * address.
+     * </blockquote></blockquote>
      */
-    repeated ServiceEndpoint gossip_endpoint = 3;
+    repeated proto.ServiceEndpoint gossip_endpoint = 3;
 
     /**
-     * A node's grpc service IP addresses and ports, IP:Port is mandatory,
-     * fully qualified domain name is optional. Maximum number of these endpoints is 8.
+     * A list of service endpoints for gRPC calls.
+     * <p>
+     * These endpoints SHALL represent the published gRPC endpoints to which
+     * clients may submit transactions.<br/>
+     * These endpoints SHOULD specify address and port.<br/>
+     * These endpoints MAY specify a DNS name.<br/>
+     * This list MUST NOT be empty.<br/>
+     * This list MUST NOT contain more than `8` entries.
+     * <p>
+     * Each network may have additional requirements for these endpoints.
+     * A client MUST check network-specific documentation for those
+     * details.
      */
-    repeated ServiceEndpoint service_endpoint = 4;
+    repeated proto.ServiceEndpoint service_endpoint = 4;
 
     /**
-     * The node's X509 certificate used to sign stream files (e.g., record stream
-     * files). Precisely, this field is the DER encoding of gossip X509 certificate.
-     * This is a mandatory field.
+     * A certificate used to sign gossip events.
+     * <p>
+     * This value MUST be a certificate of a type permitted for gossip
+     * signatures.<br/>
+     * This value MUST be the DER encoding of the certificate presented.<br/>
+     * This field is REQUIRED and MUST NOT be empty.
```

Comment:
Right now I believe all signing certs are self-signed.  I don't know if the self-signed property makes a difference on its ability to be used in the trust store for the agreement keys in mutual TLS.  We will be automatically rendering the mTLS agreement keys on boot with the signing cert as a parent certificate.  If the self-signed property is necessary for it to function as the parent cert of the mTLS key, then this should be documented here.

Reply:
Using a self-signed signing certificate as parent for the mTLS agreement key setup is neither required nor disbarred, it is just a matter of how you establish trust of the certificate and what authenticity guarantees the network is willing to accept.

That's just standard mTLS considerations.